tests/e2e/lib/setup-env.js

```diff
@@ -72,10 +79,19 @@ export const catchBeforeAll = async ( callback, errorHandler = defaultErrorHandl
 	} );
 };
 
+let video;
+const captureVideo = process.env.CAPTURE_VIDEO === 'true';
+
 async function setupBrowser() {
-	const userAgent = await browser.userAgent();
-	await page.setUserAgent( userAgent + ' wp-e2e-tests' );
-	await setBrowserViewport( 'large' );
+	const userAgent = await page.evaluate( () => navigator.userAgent );
+	logger.info( `User agent: ${ userAgent }` );
+
+	if ( captureVideo ) {
+		video = await saveVideo(
+			page,
+			path.resolve( config.get( 'testOutputDir' ), `video/video_${ new Date().getTime() }.mp4` )
```

Comment:
Is there a way to somehow link these videos to actual tests? I guess at this moment we don't know which specific test would run, but maybe in `afterAll` hook we can rename the file?

Reply:
I moved the video capture logic into jasmine reporter listeners, giving access to test names. All videos now meaningful names. I know that jasmine will be dropped at some point, but a new runner should provide the same functionality and we can move the video logic in there.